client-spark/spark2/src/main/java/org/apache/spark/shuffle/writer/RssShuffleWriter.java

```diff
@@ -247,7 +247,9 @@ public void write(Iterator<Product2<K, V>> records) {
 
   private void writeImpl(Iterator<Product2<K, V>> records) {
     List<ShuffleBlockInfo> shuffleBlockInfos;
+    int recordCount = 0;
```

Comment:
`long` will be better that should be same with the `AtomicLong`

Reply:
Done.